CONTRIBUTING.md

```diff
@@ -0,0 +1,136 @@
+# Contribution guide
+
+The codebase is maintained using the "contributor workflow" where anyone can
+contribute patch proposals using
+[pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).
+
+To contribute a patch, the workflow is as follows:
+
+1. Fork the repository.
+2. Create a topic branch.
+3. Commit patches to the branch.
+4. Push changes to the fork.
+5. Create a pull request to merge the branch of the fork into this repository.
+6. If you had someone specifically in mind, ask them to review the pull request.
+Otherwise, just wait for a code review: most members with merge permissions
+receive notifications for newly created pull requests.
+7. Address review comments, if any.
+8. Merge and submit the pull request. If you don't have merge permissions,
+a reviewer will do it for you.
+
+> **NOTE:** Before starting any coding work, please make sure to discuss the
+> issue first to agree on the optimal way of solving a particular problem.
+> If no issue exists, create one unless the change is trivial, such as
+> correcting typos and code code formatting.
+>
+> If you don't know where to start, look for issues labeled
+> with `good-first-issue`.
+
+## Branches
+
+When creating a new branch, give it a meaningful name. For example, `new-coin-xyz`.
+Later, when a pull request is finally merged in a non fast-forward way, the merge
+commit would read:
+
+    Merge branch new-coin-xyz
```

Comment:
this happens only if the person merging checks out the branch under the same name. i usually check it out under the `pr/xyz` branch name or sth.

Reply:
yeah. my point is when i see this highlighted merge commit:

![image](https://user-images.githubusercontent.com/25405/142249302-381abf6a-bd12-48c7-9555-607c5b3edade.png)

it'll take me one extra roundtrip to github to understand what that commit actually merged. compare that to "merge branch fix-passphrase-last-step" - at least i know it's something related to the passphrase

Reply:
> extra roundtrip to github

or `git log --graph ...something something`